vcell-client/src/main/java/cbit/vcell/mapping/gui/MolecularTypeSpecsTableModel.java

```diff
@@ -163,45 +190,119 @@ public Object getValueAt(int row, int col) {
 	public void setValueAt(Object aValue, int row, int col) {
 		MolecularComponentPattern mcp = getValueAt(row);
 		ColumnType columnType = columns.get(col);
+		SpeciesContextSpec scs = getSpeciesContextSpec();
+		if(scs == null) {
+			return;
+		}
+		SiteAttributesSpec sas = scs.getSiteAttributesMap().get(mcp);
 		switch (columnType) {
 		case COLUMN_SITE:
 		case COLUMN_MOLECULE:
 		case COLUMN_STATE:
 			return;
 		case COLUMN_STRUCTURE:
 			if(aValue instanceof Structure) {
-				if(getSpeciesContextSpec() == null) {
-					return;
-				}
 				Structure structure = (Structure)aValue;
-				SiteAttributesSpec sas = getSpeciesContextSpec().getSiteAttributesMap().get(mcp);
 				if(sas == null) {
-					sas = new SiteAttributesSpec(fieldSpeciesContextSpec, mcp, structure);
+					sas = new SiteAttributesSpec(scs, mcp, structure);
 				} else {
 					sas.setLocation(structure);
 				}
+				scs.firePropertyChange(SpeciesContextSpec.PROPERTY_NAME_SITE_ATTRIBUTE, null, sas);
+			}
+			return;
+		case COLUMN_X:
+			if (aValue instanceof String) {
+				if(sas == null) {
+					sas = new SiteAttributesSpec(scs, mcp, scs.getSpeciesContext().getStructure());
+				}
+				String newExpressionString = (String)aValue;
+				double res = 0.0;
+				try {
+					res = Double.parseDouble(newExpressionString);
+				} catch(NumberFormatException e) {
+					return;
```

Comment:
does this behave properly if you type something bad for column_x,y,z?  it seems to ignore things, but it doesn't throw an exception to reject a bad column ... will it just silently not update and not notify the user?

Reply:
I figured this will fail as a result of a typo, most likely the user will instantly figure out what went wrong. I didn't want to bother him with an exception dialog he'll have to close